### PR TITLE
tests: unit: util: Rework test structure for C++ testing

### PR DESCRIPTION
**Pull request by nordic-krch**
Rework test structure to handle C++ test in a better way. Similar approach already applied in other tests which are executed for C and C++. Getting rid of test.inc which due to non-standard extension was not covered by complience check.

### CODE REVIEW

**Comment by nordic-krch** [comment]
Example of similar rework: #43791

**Comment by nordic-krch** [comment]
It seems that checkpatch is expecting that `FOR_EACH()` macro is a conditional statement like `for` or `switch`. @keith-zephyr can you help?

**Comment by keith-zephyr** [comment]
checkpatch is confused because it doesn't recognize that `FOR_EACH_NONEMPTY_TERM` includes a terminating comma and wants the following lines indented as if was a line continuation.

Adding unnecessary indents stops checkpatch from complaining (but I'm not suggesting you do this)
```
	uint8_t array[] = {
		FOR_EACH_NONEMPTY_TERM(SQUARE, (,))
			FOR_EACH_NONEMPTY_TERM(SQUARE, (,),)
				255
	};
```

I think the checkpatch configuration is attempting to detect code blocks defined by some of the [single](https://github.com/zephyrproject-rtos/zephyr/blob/1dc6279d12ea81041c21c8e76d6eae6a2e17bba0/include/zephyr/sys/slist.h#L57)/[double](https://github.com/zephyrproject-rtos/zephyr/blob/1dc6279d12ea81041c21c8e76d6eae6a2e17bba0/include/zephyr/sys/dlist.h#L109C13-L109C13) link list macros.

The fix probably involves generating an allow  list of FOR_EACH macros that define a code block.  The alternative to create a list of FOR_EACH macros that should be ignored by this check.

**Comment by nordic-krch** [comment]
@nashif @yperess @tejlmand can you take a look? @keith-zephyr can you confirm if checkpatch patch is ok?

**Review by nashif — DISMISSED** [review]

File: scripts/checkpatch.pl [hunk 1]
```diff
@@ -3739,7 +3739,7 @@ sub process {
 
 # if/while/etc brace do not go on next line, unless defining a do while loop,
 # or if that brace on the next line is for something else
-		if ($line =~ /(.*)\b((?:if|while|for|switch|(?:[A-Z_]+|)FOR_EACH[A-Z_]+)\s*\(|do\b|else\b)/ && $line !~ /^.\s*\#/) {
+		if ($line =~ /(.*)\b((?:if|while|for|switch|(?:[A-Z_]+|)FOR_EACH(?!_NONEMPTY_TERM|_IDX|_FIXED_ARG|_IDX_FIXED_ARG)[A-Z_]+)\s*\(|do\b|else\b)/ && $line !~ /^.\s*\#/) {
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply scripts/checkpatch.pl]: removed those from that condition.

**Review by keith-zephyr — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@nordic-krch hey this needs a rebase

**Comment by nordic-krch** [comment]
@nashif @keith-zephyr I had resolve conflicts can you re-approve?

**Review by nashif — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**